src/main/java/com/mongodb/hibernate/internal/translate/AbstractMqlTranslator.java

```diff
@@ -301,7 +320,104 @@ public void visitParameter(JdbcParameter jdbcParameter) {
 
     @Override
     public void visitSelectStatement(SelectStatement selectStatement) {
-        throw new FeatureNotSupportedException("TODO-HIBERNATE-22 https://jira.mongodb.org/browse/HIBERNATE-22");
+        logSqlAst(selectStatement);
+        if (!selectStatement.getQueryPart().isRoot()) {
+            throw new FeatureNotSupportedException("Subquery not supported");
+        }
+        if (!selectStatement.getCteStatements().isEmpty()
+                || !selectStatement.getCteObjects().isEmpty()) {
+            throw new FeatureNotSupportedException("CTE not supported");
+        }
+        selectStatement.getQueryPart().accept(this);
+    }
+
+    @Override
+    public void visitQuerySpec(QuerySpec querySpec) {
+        if (!querySpec.getGroupByClauseExpressions().isEmpty()) {
+            throw new FeatureNotSupportedException("GroupBy not supported");
+        }
+        if (querySpec.hasSortSpecifications()) {
+            throw new FeatureNotSupportedException("Sorting not supported");
+        }
+
+        var collection = acceptAndYield(querySpec.getFromClause(), COLLECTION_NAME);
+
+        var whereClauseRestrictions = querySpec.getWhereClauseRestrictions();
+        var filter = whereClauseRestrictions == null || whereClauseRestrictions.isEmpty()
```

Comment:
Can `whereClauseRestrictions ` ever be null in practice?  If not, we can use an assertion instead of the null check.

Reply:
It seems perfectly fine to be null (e.g. `select b from Book b` with pagination constraint either in HQL per se or query options).

The above pattern was borrowed from [AbstractSqlTranslator](https://github.com/hibernate/hibernate-orm/blob/main/hibernate-core/src/main/java/org/hibernate/sql/ast/spi/AbstractSqlAstTranslator.java#L3727) as below:
```
protected final void visitWhereClause(Predicate whereClauseRestrictions) {
		final Predicate additionalWherePredicate = this.additionalWherePredicate;
		if ( whereClauseRestrictions != null && !whereClauseRestrictions.isEmpty() || additionalWherePredicate != null ) {
			appendSql( " where " );

	        ... ...
```